mentat/git_handler.py

```diff
@@ -1,10 +1,13 @@
 import logging
 import os
 import subprocess
+from contextvars import ContextVar
 from pathlib import Path
 
 from mentat.errors import UserError
 
+GIT_ROOT: ContextVar[Path] = ContextVar("mentat:git_root")
```

Comment:
Should we use GIT_ROOT directly in the functions below?

Reply:
Yes, thanks for finding this! I probably missed a few other places I should have used the contextvar haha